Sources/Property.swift

```diff
@@ -571,27 +571,28 @@ public final class Property<Value>: PropertyProtocol {
 					return observer.action(event)
 				}
 
-				box.lock.sync {
-					if let value = event.value {
-						box.value = value
+				box.modifyDidSet({ value in
+					if let newValue = event.value {
+						value = newValue
 					}
+				}, didSet: {
 					observer.action(event)
```

Comment:
Can't `observer.action` be in the `action` block? It still happens inside the lock either way.